src/simplewallet/simplewallet.cpp

```diff
@@ -1789,47 +1825,84 @@ bool simple_wallet::refresh(const std::vector<std::string>& args)
   return refresh_main(start_height, false);
 }
 //----------------------------------------------------------------------------------------------------
-bool simple_wallet::show_balance_unlocked()
+bool simple_wallet::show_balance_unlocked(bool detailed)
 {
+  success_msg_writer() << tr("Currently selected account: [") << m_wallet->get_current_subaddress_account() << tr("] ") << m_wallet->get_subaddress_label(0);
   success_msg_writer() << tr("Balance: ") << print_money(m_wallet->balance()) << ", "
     << tr("unlocked balance: ") << print_money(m_wallet->unlocked_balance());
+  std::map<uint32_t, uint64_t> balance_per_subaddress = m_wallet->balance_per_subaddress();
+  std::map<uint32_t, uint64_t> unlocked_balance_per_subaddress = m_wallet->unlocked_balance_per_subaddress();
+  if (!detailed || balance_per_subaddress.empty())
+    return true;
+  success_msg_writer() << tr("Balance per address:");
+  success_msg_writer() << boost::format("%15s %21s %21s %21s %7s") % tr("Address") % tr("Balance") % tr("Unlocked balance") % tr("Label") % tr("Outputs");
+  std::vector<tools::wallet2::transfer_details> transfers;
+  m_wallet->get_transfers(transfers);
+  for (auto& i : balance_per_subaddress)
+  {
+    std::string address_str = m_wallet->get_subaddress_as_str(i.first).substr(0, 6);
+    cryptonote::subaddress_index subaddr_index = {m_wallet->get_current_subaddress_account(), i.first};
+    uint64_t num_unspent_outputs = std::count_if(transfers.begin(), transfers.end(), [&](const tools::wallet2::transfer_details& td) { return !td.m_spent && td.m_subaddr_index == subaddr_index; });
+    success_msg_writer() << boost::format(tr("%8u %6s %21s %21s %21s %7u")) % i.first % address_str % print_money(i.second) % print_money(unlocked_balance_per_subaddress[i.first]) % m_wallet->get_subaddress_label(i.first) % num_unspent_outputs;
```

Comment:
You don't say whether you know for a fact it's initialized to 0. A uint64_t ctor doesn't necessarily mean it's 0, since a function scope "uint64_t x;" won't init x to 0.